### DISCUSSION TITLE
Using the Python DagsterGraphQLClient against a Dagster Cloud instance

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#7772] [Q&A] [answered]
The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1651616623616629?thread_ts=1651616623.616629&cid=C01U954MEER). Reposting it here for posterity and discoverability.

### Original question:

is there an example somewhere of pointing the python DagsterGraphQLClient to a remote dagster instance (i.e. Dagster cloud)? It's unclear to me how I would provide authentication

**Comment by prha** [comment, marked as answer]
Here's how it works on Dagster Cloud:

```
from dagster_graphql import DagsterGraphQLClient
url = "yourorg.dagster.cloud/prod"
token = "your_token_here" # a User Token generated from the Cloud Settings page in Dagster Cloud. Note: User Token, not Agent Token
client = DagsterGraphQLClient(url, use_https=True, headers={"Dagster-Cloud-Api-Token": token})
```

**Reply by markfickett** [reply]
Thanks, this was useful to me. I'm looking to streamline the docker build, tag, push, and reload location workflow, so being able to trigger a reload via GraphQL is handy, and having it wrapped up in the Python GraphQL client is great.

I had to change:
```py
url=url="/graphql"
```
to
```py
url=f"https://{url}/graphql"
```

**Reply by gibsondan** [reply]
Updated the answer with that change, thanks!

**Comment by perryrjohnson** [comment]
Is this still the preferred way to pointing the python DagsterGraphQLClient to a remote dagster instance? having issues passing in auth to connect and make this work with `1.0.7`

**Reply by gibsondan** [reply]
Hi Perry - this way should still work, could you share more about the issues you’re seeing?

**Reply by perryrjohnson** [reply]
hey there! I'm trying this to kick off a submit job execution:

```from dagster_graphql import DagsterGraphQLClient
from gql.transport.requests import RequestsHTTPTransport

org_name = ""
url = f"https://{org_name}.dagster.cloud/prod"
token = "" # a User Token generated from the Cloud Settings page in Dagster Cloud. Note: User Token, not Agent Token
client = DagsterGraphQLClient(hostname=url, transport=RequestsHTTPTransport(url=f"https://{url}/graphql", headers={"Dagster-Cloud-Api-Token": token}))

client.submit_job_execution(job_name="all_assets_job", 
                            repository_location_name="example_location", 
                            repository_name="my_dagster_project")
```

and I'm getting this error:
```
NewConnectionError: <urllib3.connection.HTTPSConnection object at 0x10f5f57d0>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known```

**Reply by perryrjohnson** [reply]
Just not sure if I'm passing in the tokens the wrong way or how to check if that client object is connected successfully. thanks

**Reply by gibsondan** [reply]
I think you have “https” in there twice in the constructor to RequestsHTTPTransport - once from the variable and once again in the string

**Reply by perryrjohnson** [reply]
ugh. (face palm) yes. that would do it. thanks.
